app/services/conviction_decision_tree.rb

```diff
@@ -76,11 +84,27 @@ def after_conviction_length_type
   end
 
   def after_compensation_paid
-    return edit(:compensation_payment_date) if GenericYesNo.new(disclosure_check.compensation_paid).yes?
+    return edit(:compensation_paid_amount) if GenericYesNo.new(disclosure_check.compensation_paid).yes?
 
     show(:compensation_not_paid)
   end
 
+  def after_compensation_paid_amount
+    edit(:compensation_payment_date)
```

Comment:
As mentioned in slack, in this case, you can just get rid of this method and add the `edit(:compensation_payment_date)` directly inside the case branch. We do that usually when the steps are lineal, with no conditions when moving from one to another.